extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -280,8 +283,8 @@ public void runActionInTransaction(
           LOGGER.debug("transaction committed");
         }
       } catch (Exception e) {
+        LOGGER.debug("Rolled back transaction due to an error", e);
         tr.rollback();
```

Comment:
If this rollback fails, we never saw any logging previously.

Reply:
Agreed this is not in scope for this PR, but maybe we could consider it. As long as we maintain transactionality, anything is ok. 

However my guess is that most (99+%) of failures will be non-recoverable. Before changing this logic, I would advocate for us collecting stats on recoverable failures and perhaps trying to tailor the retry to only recoverable failures.

Reply:
Currently, we would retry if OptimisticLockException is hit - concurrent access. Yeah, we can collect what can be retried here. I remember the caller will do retry. Is that right?

Reply:
In my tests, no the caller did not retry if this query failed.